File: Segundo_Semestre/Programacao_Orientada_Objeto/aula_05_heranca/src/Administrativo.java
import java.util.Scanner;

public class Administrativo extends Funcionario{

    private String departamento;

    public Administrativo() {}

    public Administrativo(int matricula, String nome, float salario, String departamento) {
        super(matricula, nome, salario);
        setDepartamento(departamento);
    }

    public String getDepartamento() {
        return departamento;
    }

    public void setDepartamento(String departamento) {
        this.departamento = departamento;
    }

    public String toString(){
        String saida = super.toString();
        saida += "\nDepartamento: " + getDepartamento();
        return saida;
    }

    public void setStrig(){
        Scanner scan = new Scanner(System.in);
        super.setStrig();
        System.out.print("Digite o Departamento: ");
        String departamento = scan.nextLine();
        setDepartamento(departamento);
    }
}

File: Terceiro_Semestre/Desenvolvimento_Mobile/aula_05_conceitos_classe/exemplo_herancas/src/Administrativo.kt
class Administrativo : Funcionario {
    var departamento: String = ""
        set(value) {
            if (value != "") field = value
        }
        get() {
            return if (field != "") field else "Departamento não cadastrado. "
        }
    constructor(mat: Int, nome: String, sal: Double, depto: String) : super(mat, nome, sal){
        departamento = depto
    }

    override fun toString() : String{
        return ("\nMatrícula: $matricula"
                + "\nNome.....: $nome"
                + "\nSalário..: $salario"
                + "\nDepartamento....: $departamento")
    }

}